internal/clients/http/sources/sources_client.go

```diff
@@ -107,51 +115,55 @@ func (c *SourcesClient) ListProvisioningSources(ctx context.Context) (*[]clients
 }
 
 func (c *SourcesClient) GetArn(ctx context.Context, sourceId clients.ID) (string, error) {
-	ctxval.Logger(ctx).Info().Msgf("Getting ARN of source %v", sourceId)
+	logger := ctxval.Logger(ctx)
+	logger.Trace().Msgf("Getting ARN of source %v", sourceId)
+
 	// Get all the authentications linked to a specific source
 	resp, err := c.client.ListSourceAuthenticationsWithResponse(ctx, sourceId, &ListSourceAuthenticationsParams{}, headers.AddSourcesIdentityHeader)
 	if err != nil {
 		return "", fmt.Errorf("cannot list source authentication: %w", err)
 	}
-	statusCode := resp.StatusCode()
-	if parsing.IsHTTPNotFound(statusCode) {
-		return "", NewNotFoundError(ctx, "authentications for source weren't found in sources app")
-	}
-	if !parsing.IsHTTPStatus2xx(statusCode) {
-		ctxval.Logger(ctx).Warn().Msgf("Sources replied with unexpected status while fetching Authentications per source: %v", statusCode)
-		return "", SourcesClientErr
+
+	err = http.HandleHTTPResponses(ctx, resp.StatusCode())
+	if err != nil {
+		if errors.Is(err, clients.NotFoundError) {
+			return "", SourceNotFoundErr
```

Comment:
I'd really love, if all our clients would throw the same errors, at least wrapped? I liked it the way linter did not xD

Reply:
Yes I wrapped all of them.

Reply:
you are creating a `clients.NotFoundError` that is very useful to the outside world, but then you drop it here completely and return `sources.SourceNotFoundErr` that is IMHO completely useless to outside world (outside of http package)